### PR TITLE
worldmap: add RC altar Icons

### LINKED ISSUE #3736: Show Runecrafting altars on the map
It would be nice to know where the Runecrafting altars are when looking at the minimap. I suggest placing a marker on each mysterious ruins to indicate which altar it is.

### LINKED ISSUE #9758: Improvement suggestion: Add RC altar icons to the RuneScape surface world map
We have icons in Abyss on the radar indicating to which runecrafting altar each exit leads. Could we perhaps add the same icons on the world map and radar indicating the location of mysterious ruins resp. blood & soul altars? This could help new players and probably easier clicking on the correct location when making combo runes.

### PR DESCRIPTION
**Pull request by Dava96**
closes #9758
closes #3736

### CODE REVIEW

**Comment by oplosthee** [comment]
I feel like the water rune icon could lead to some confusion as it looks a lot like the fountain/water source icon on the worldmap. The blood rune icon also looks a lot like the 💩 emoji, maybe a red recolor of the water icon, like is done with runes in-game, might look a bit better?

**Comment by raiyni** [comment]
I'm not sure if these were drawn by hand or resized down to size but you should try getting the icons more centered and removing the color artifacting. It might make astrals and air runes easier to see.

File: runelite-client/src/main/java/net/runelite/client/plugins/worldmap/WorldMapPlugin.java [hunk 1]
```diff
@@ -301,7 +303,17 @@ private void updateShownIcons()
...
+			worldMapPointManager.removeIf(RunecraftingAltarPoint.class::isInstance);
+
+			if (config.runecraftingAltarIcon())
+			{
+				Arrays.stream(RunecraftingAltarLocation.values())
+					.filter(altarLocation -> altarLocation.getLocation() != null)
+					.map(altarLocation -> new RunecraftingAltarPoint(altarLocation, BLANK_ICON))
+					.forEach(worldMapPointManager::add);
+			}
+		}
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldmap/WorldMapPlugin.java]: All of this should be de-indented by one tab.

File: runelite-client/src/main/java/net/runelite/client/plugins/worldmap/WorldMapPlugin.java [hunk 2]
```diff
@@ -301,7 +303,17 @@ private void updateShownIcons()
 				}
 			}).map(TeleportPoint::new)
 			.forEach(worldMapPointManager::add);
-	}
+
+			worldMapPointManager.removeIf(RunecraftingAltarPoint.class::isInstance);
+
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldmap/WorldMapPlugin.java]: This blank line can be removed, as clearing RC altar points is directly related to the subsequent adding of them.
```suggestion
```

File: runelite-client/src/main/java/net/runelite/client/plugins/worldmap/WorldMapPlugin.java [hunk 3]
```diff
@@ -301,7 +303,17 @@ private void updateShownIcons()
...
+			if (config.runecraftingAltarIcon())
+			{
+				Arrays.stream(RunecraftingAltarLocation.values())
+					.filter(altarLocation -> altarLocation.getLocation() != null)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldmap/WorldMapPlugin.java]: Will there ever be altars with null locations? There aren't currently any like that in the added enum.

**Pull request closed without merging**